survey-sdk/src/main/java/co/nimblehq/survey/sdk/network/NetworkBuilder.kt

```diff
@@ -65,21 +74,33 @@ abstract class NetworkBuilder {
             .baseUrl(baseUrl)
             .addConverterFactory(
                 getJsonApiConverterFactory(
-                    provideJsonApiMoshi(
-                        Moshi.Builder(),
-                        provideJsonApiFactory()
+                    provideMoshiJsonApi(
+                        provideJsonApiFactory(),
+                        Moshi.Builder()
                     )
                 )
             )
             .addConverterFactory(getConverterFactory(Moshi.Builder()))
     }
 
-    protected fun buildAppService(): AppService {
-        return provideRetrofit().create(AppService::class.java)
-    }
-
     inline fun <reified T> buildService(): T {
         return provideRetrofit().create(T::class.java)
     }
 
+    inner class TokenInterceptor : Interceptor {
+        @Throws(IOException::class)
+        override fun intercept(chain: Interceptor.Chain): Response {
+            val originalRequest = chain.request()
+            val authenticationRequest = request(originalRequest)
+            return chain.proceed(authenticationRequest)
+        }
+
+        private fun request(originalRequest: Request): Request {
+            val builder = originalRequest.newBuilder()
+            if (token.isNotEmpty()) {
+                builder.addHeader("Authorization", "$tokenType $token")
```

Comment:
OPTIONAL: `"Authorization"`  could be defined as a constant

Reply:
@hoangnguyen92dn Agree.